biapol_utilities/utilities/_higher_dimensions.py

```diff
@@ -0,0 +1,84 @@
+"""Apply function to images owning higher dimensions than designed input."""
+import numpy as np
+import dask
+import dask.array as da
+
+
+# Adapted from https://stackoverflow.com/a/7663441/11885372
+def _get_diff_in_dimensions(func, shape, *args, **kwargs):
+    # Same shape as image
+    shape = tuple([1]*len(shape))
+    dummy_image = np.ones(shape)
+    for attempt in range(10):
+        try:
+            func(dummy_image, *args, **kwargs)
+        except Exception as e:
+            dummy_image = dummy_image[0]
+        else:
+            return(attempt)
+    return(None)
+
+
+def _apply_nD_func_to_nplus1_D_image(image, func, *args, **kwargs):
+    """Apply function whose input is a nD image to a (n+1) image using dask."""
+    # List of lazy (to be) processed images
+    list_of_dask = [
+                    dask.delayed(func)(image[i],
+                                       *args, **kwargs)
+                    for i in range(image.shape[0])
+    ]
+    # List of lazy dask arrays
+    dask_arrays = [
+        da.from_delayed(lazy_array, shape=image[0].shape, dtype=image[0].dtype)
+        for lazy_array in list_of_dask
+    ]
+    # List to dask stack
+    dask_stack = da.stack(dask_arrays, axis=0)
+    return(dask_stack)
+
+
+def process_higher_dimension(image, function, function_input_dimension=None,
```

Comment:
Maybe, we can have a short chat about naming this function. When reading "process higher dimension" I don't think of image processing. In ImageJ2, we call a sub-stack with n < m dimensions taken from a stack with m dimensions a "hyperslice". Also the term "hyperstack" is quite common in ImageJ land for stacks with more than three dimensions. Last but not least, a function that applies function to data, is in python called ["map"](https://docs.python.org/3/library/functions.html#map). Maybe we can have a name that fits into this scheme? "map_hyperslices()", or we use the dask terminology and call it something like "map_hyperblocks" [See also](https://docs.dask.org/en/latest/generated/dask.array.map_blocks.html).

Also @jo-mueller do you have opinions?

Reply:
So...do I understand correct that these functions are designed to detect when an image is too high-dimensional for the function to be applied? This could be an instance where it could make sense to use recursive function definitions, i.e. do something like:

```Python
def some_function(image, param):
   if image.ndim > 3:
      output = some_function(image[0])
```